### PR TITLE
Implement `colorPalette` strategy for composability

### PR DESCRIPTION
**Pull request by hobbescodes**
##### Task link: https://trello.com/c/Kk74Oh47/328-update-all-components-to-use-https-panda-csscom-docs-concepts-virtual-color-colorpalette-%F0%9F%A6%A4

Updated all `core` components to use `colorPalette` to define standard for styles that involve color.

## Test Steps

1) Verify that logic is sound.
2) Verify that all components render appropriately in SB.
3) Verify that default colors are appropriate.
4) Verify that changing `colorPalette` prop at root level for each component updates styles appropriately.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,6 +107,7 @@
     "next": "^14.0.4",
     "postcss": "^8.4.32",
     "prettier": "^3.1.1",
+    "radash": "^11.0.0",
```
Comment (coopbri) [inline package.json:110]: Is `radash` used anywhere?

File: src/lib/panda/colors.ts [hunk 2]
```diff
@@ -1,62 +1,506 @@
...
-      900: { value: "#a87943" },
+import { hexToRgba } from "../util";
+
+import type { Tokens } from "@pandacss/dev";
+import type { Recursive, Token } from "@pandacss/types";
+
+/**
+ * Generate alpha (transparent) colors from base colors.
+ */
+export const generateAlphaColors = (
```
Comment (coopbri) [inline src/lib/panda/colors.ts:12]: this looks oddly familiar

Comment (hobbescodes) [reply src/lib/panda/colors.ts:12]: Custom logic sir.... creator is [redacted]

Comment (coopbri) [reply src/lib/panda/colors.ts:12]: God

</html>

**Review by coopbri — APPROVED** [review]

**Pull request merged**